anima-jws-voucher.mkd

```diff
@@ -55,95 +55,101 @@ informative:
 For example, {{BRSKI}} provides this context via the media type for the voucher artifact.
 This document utilizes the optional "typ" (Type) Header Parameter of JWS {{RFC7515}} to provide information about the signed object.
 
-This document should be considered an update to {{RFC8366}} in the category of "See Also"
-as per {{?I-D.kuehlewind-update-tag}}.
+This document should be considered an update to {{I-D.draft-ietf-anima-rfc8366bis}} in the category of "See Also" as per {{?I-D.kuehlewind-update-tag}}.
 TODO: double check with RFC8366bis
 
 # Terminology
 
 {::boilerplate bcp14}
 
+Voucher:
+: A signed statement from the MASA service {{I-D.draft-ietf-anima-rfc8366bis}}.
+
+JWS Voucher:
+: A JWS structure signing the JSON Voucher Data.
+
+JSON Voucher Data:
+: An unsigned JSON document that conforms with the data model described by the ietf-voucher YANG module.
+
+BASE64URL(OCTETS):
+: Denotes the base64url encoding of OCTETS, per {{Section 2 of RFC7515}}.
+
+UTF8(STRING):
+: Denotes the octets of the UTF-8 {{RFC3629}} representation of STRING, per {{Section 1 of RFC7515}}.
+
 # Voucher Artifact with JSON Web Signature
 
 {{RFC7515}} defines the following serializations for JWS:
 
 1. "JWS Compact Serialization" in {{Section 7.1 of RFC7515}}
 2. "JWS JSON Serialization" in {{Section 7.2 of RFC7515}}
-- "General JWS JSON Serialization Syntax" in {{Section 7.2.1 of RFC7515}}   
+- "General JWS JSON Serialization Syntax" in {{Section 7.2.1 of RFC7515}}
 - "Flattened JWS JSON Serialization Syntax" in {{Section 7.2.2 of RFC7515}}
 
-This document makes use of the "General JWS JSON Serialization Syntax" to support multiple signatures,
+This document makes use of the "General JWS JSON Serialization Syntax" of {{RFC7515}} to support multiple signatures,
 as already supported by {{RFC8366}} for CMS-signed vouchers.
 
-The {{RFC8366}} voucher data structure consists of a nested map, the outer map of which is:
-
-~~~~
-    { "ietf-voucher:voucher" : { inner map }}
-~~~~
-
-This outer map is considered the JWS Payload as described in {{Section 3 of RFC7515}}.
-A "JWS JSON Serialization Overview" is given in {{Section 3.2 of RFC7515}} and more details on the JWS serializations in {{Section 7 of RFC7515}}.
-
 ## Voucher Representation in General JWS JSON Serialization Syntax
 
-The following figure gives an overview of the Voucher representation in "General JWS JSON Serialization Syntax":
+JWS Voucher artifacts MUST use the "General JWS JSON Serialization Syntax" defined in {{Section 7.2.1 of RFC7515}}.
+A "JWS JSON Serialization Overview" is given in {{Section 3.2 of RFC7515}} and more details on the JWS serializations in {{Section 7 of RFC7515}}.
+The following figure summarizes the serialization of JWS Voucher artifacts:
 
 ~~~~
     {
-      "payload": "BASE64URL(ietf-voucher:voucher)",
+      "payload": BASE64URL(UTF8(JSON Voucher Data)),
```

Comment:
For me `ietf-voucher:voucher`does not identify what is meant here.
If apply JSON logic, this member key would only give me the inner map, but we need the outer map having the full member with key and value.
In my opinion, the main issue is that there is no proper term defined for this unsigned JSON structure, hence I recommend to define "JSON Vocher Data" (to be discussed if "Data" can be dropped) already in RFC8366 -- see my longer comment.

Reply:
https://mailarchive.ietf.org/arch/msg/anima/nFdBd102-qppS0oO6JuYmabtj1w/